domains/client/domain-operator/src/bootstrapper.rs

```diff
@@ -0,0 +1,101 @@
+use futures::StreamExt;
+use sc_client_api::{BlockchainEvents, ImportNotifications};
+use sp_api::{HeaderT, NumberFor, ProvideRuntimeApi};
+use sp_blockchain::HeaderBackend;
+use sp_domains::{DomainId, DomainInstanceData, DomainsApi, DomainsDigestItem};
+use sp_runtime::traits::Block as BlockT;
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+pub struct BootstrapResult<CBlock: BlockT> {
+    // The `DomainInstanceData` used by the domain instance starter to
+    // construct `RuntimeGenesisConfig` of the domain instance
+    pub domain_instance_data: DomainInstanceData,
+    // The `imported_block_notification_stream` used by the bootstrapper
+    //
+    // NOTE: the domain instance starter must reuse this stream instead of
+    // create a new one from the consensus client to avoid missing imported
+    // block notification.
+    pub imported_block_notification_stream: ImportNotifications<CBlock>,
+}
+
+/// Domain instance bootstrapper
+pub struct Bootstrapper<Block, CBlock, CClient> {
+    consensus_client: Arc<CClient>,
+    _phantom_data: PhantomData<(Block, CBlock)>,
+}
+
+impl<Block, CBlock, CClient> Bootstrapper<Block, CBlock, CClient>
+where
+    Block: BlockT,
+    CBlock: BlockT,
+    CClient: HeaderBackend<CBlock>
+        + ProvideRuntimeApi<CBlock>
+        + BlockchainEvents<CBlock>
+        + Send
+        + Sync
+        + 'static,
+    CClient::Api: DomainsApi<CBlock, NumberFor<Block>, Block::Hash>,
+{
+    pub fn new(consensus_client: Arc<CClient>) -> Self {
+        Bootstrapper {
+            consensus_client,
+            _phantom_data: Default::default(),
+        }
+    }
+
+    pub async fn run(
```

Comment:
`fetch_domain_bootstrap_info` might be more approariate than `run`. I would add a comment if not as this is not going to be long running since `run` implies long running